fetchData.js

```diff
@@ -0,0 +1,98 @@
+// first grab the search form
+const searchForm = document.querySelector("form");
+// now catch the search result div
+const searchResultDiv = document.querySelector(".search-result");
+//grab the entire container:
+const container = document.querySelector(".container");
+
+const API_Key = "AIzaSyB2CL69PUQsH4j4lRmMFJmm07nE1dB8rTE";
+
+//Pagination:
+const Pagination = document.querySelector(".pagination");
+let start_video_value = 0;
+let last_video_value = 4;
+
+let searchQuery = "";
+//Function to catch what's in the search query:
+searchForm.addEventListener("submit", (e) => {
+  e.preventDefault();
+  searchQuery = e.target.querySelector("input").value;
+  //write the fetch API function: Because we are fetching it, we use async
+  fetchAPI();
```

Comment:
We generally fetch data not the API; So I think function name should be like fetch videos by passing the specific URL/endpoint;